Add "vckick" example to miscellaneous-examples.md

apri1l opened the pull request:
I've put together a new example for the miscellaneous examples page called "Kicking users (or bots) from a voice channel". It's a simple bit of code that can be used to kick users (and bots) from a voice channel, and it's pretty quick, only relying on 2 permissions as well. 😄

apri1l committed:
Update miscellaneous-examples.md, "vckick" example

apri1l committed:
"vckick" update: Make it consistent

apri1l commented:
Also, I'm so sorry if I've done the pull request incorrectly, I've literally just made a GitHub account to submit this example, so it's a first.

apri1l committed:
Change "&&" to "||" in the check for permissions

Stitch07 commented on examples/miscellaneous-examples.md:
```diff
@@ -217,3 +217,31 @@ if( swearWords.some(word => message.content.includes(word)) ) {
 
+### Kicking users (or bots) from a voice channel
+
+> Made by `April#8035`
```
This is unnecessary and doesn't follow the style of the other guides. All contents are presumably made by a contributor so it isn't needed.

apri1l replied:
Removed.

Stitch07 commented on examples/miscellaneous-examples.md:
```diff
@@ -217,3 +217,31 @@ if( swearWords.some(word => message.content.includes(word)) ) {
+
+Being able to kick a user or bot from a voice channel doesn't come within Discord sadly, and it's a great feature to have for general use, or if your developing a music bot or related that needs to be removed quickly. Luckily, there's a simple and fast way to do it, and it can be added easily.
+
+**This example was made for discord.js-master, AKA v12, and it was designed with async/await usage in mind. It should be rather easy to backport to v11/stable however.**
```
v12 isn't the stable version and the rest of the guide follows v11.x (npm version). At this point all guides should be written following v11.

apri1l replied:
Converted back to v11.

Stitch07 commented on examples/miscellaneous-examples.md:
````diff
@@ -217,3 +217,31 @@ if( swearWords.some(word => message.content.includes(word)) ) {
+
+```javascript
+// Make sure the bot user has permissions to make channels and move members in the guild:
+if (!message.guild.me.hasPermission('MANAGE_CHANNELS') && !message.guild.me.hasPermission('MOVE_MEMBERS')) return message.reply('missing the required `Manage Channels` and `Move Members` permissions.');
````
This can be made ```message.guild.me.hasPermission(['MANAGE_CHANNELS', 'MOVE_MEMBERS']);```

apri1l replied:
Changed over to using that code.

Stitch07 commented on examples/miscellaneous-examples.md:
```diff
@@ -217,3 +217,31 @@ if( swearWords.some(word => message.content.includes(word)) ) {
+if (!message.guild.me.hasPermission('MANAGE_CHANNELS') && !message.guild.me.hasPermission('MOVE_MEMBERS')) return message.reply('missing the required `Manage Channels` and `Move Members` permissions.');
+
+// Get the mentioned user/bot and check if they're in a voice channel:
+const user = message.mentions.members.first();
```
For readability this should be `member`, as it returns a GuildMember not a User.

apri1l replied:
Changed that over to `member`.

Stitch07 commented on examples/miscellaneous-examples.md:
```diff
@@ -217,3 +217,31 @@ if( swearWords.some(word => message.content.includes(word)) ) {
+
+// Get the mentioned user/bot and check if they're in a voice channel:
+const user = message.mentions.members.first();
+if (!user) return message.reply('you need to @mention a user/bot to kick from the voice channel.');
```
The reply should be in title case.

apri1l replied:
Changed the messages all over to title case.

Stitch07 commented on examples/miscellaneous-examples.md:
```diff
@@ -217,3 +217,31 @@ if( swearWords.some(word => message.content.includes(word)) ) {
+// Get the mentioned user/bot and check if they're in a voice channel:
+const user = message.mentions.members.first();
+if (!user) return message.reply('you need to @mention a user/bot to kick from the voice channel.');
+if (!user.voiceChannel) return message.reply('that user/bot isn\'t in a voice channel.');
```
Looks good otherwise.

apri1l replied:
👍

apri1l committed:
Remove "contributor" quote

apri1l committed:
Convert back to discord.js v11/stable

apri1l committed:
Updated permission check, use title case

apri1l committed:
Change "user" to "member"

kyranet reviewed (commented):
The code looks fine to me, although I'm honestly not very used to discord.js@11.3.x code.

I have one suggestion, in the temporary voice channel, assign permissions to deny @everyone's permissions to view it (but allow it for the target user). That way, when you voice kick somebody, nobody sees the new channel but the user getting voicekicked and the guild administrators.

You can achieve that by adding the permissions to the create channel's options.

apri1l commented:
Sounds fine, I'll modify it so the `@everyone` role can't see the new channel.

apri1l committed:
Add permissions overrides 

Only the user/bot that's being kicked and the guild administrators/owner can see and use the channel now, as requested.

eslachance commented:
Here's another comment : make sure you add that the message event must be async. This might not be obvious to everyone and they'll come crying `but but unexpected keyword await!` in the server.

apri1l committed:
Add information about async/await

apri1l commented:
Added that, actually a good point seeing how I complained about that when I was a beginner.
